services/payments/statemachine_solana.go

```diff
@@ -202,18 +204,32 @@ func (sm *SolanaMachine) Pay(ctx context.Context) (*paymentLib.AuthenticatedPaym
 			idempotencyData.Transaction.Signatures,
 		)
 	}
+	logger.Debug().Str("documentID", sm.transaction.DocumentID).Msg("loaded external idempotency data for transaction")
 
 	b58Signature := base58.Encode(idempotencyData.Transaction.Signatures[0])
+	// solanaError stores temporary errors that need to be recorded as the LastError in QLDB but
+	// that are expected and do not need to be acted upon. It will only be returned if no other
+	// unexpected errors arise from processing.
+	var solanaError error
 	status, err := checkStatus(ctx, b58Signature, &sm.solanaRpcClient)
 	if err != nil {
-		return sm.transaction, fmt.Errorf("failed to check transaction status: %w", err)
+		// Some errors are expected and we just want to record them. Check if the status check
+		// returned such an error and prepare to return it after we attempt to make progress.
+		if errors.Is(err, SolanaTransactionNotConfirmedError) ||
+			errors.Is(err, SolanaTransactionNotFoundError) ||
+			errors.Is(err, SolanaTransactionUnknownError) {
```

Comment:
This could also be extracted into a helper predicate:

```golang
func isErrSolanaTransaction(err error) bool {
	if err == nil {
		return false
	}
	
	return errors.Is(err, SolanaTransactionNotConfirmedError) || errors.Is(err, SolanaTransactionNotFoundError) || errors.Is(err, SolanaTransactionUnknownError)
}
```